features/lorawan/LoRaWANInterface.h

```diff
@@ -75,6 +75,37 @@ class LoRaWANInterface: public LoRaWANBase {
     virtual lorawan_status_t get_rx_metadata(lorawan_rx_metadata &metadata);
     virtual lorawan_status_t get_backoff_metadata(int &backoff);
     virtual lorawan_status_t cancel_sending(void);
+    lorawan_time_t get_current_gps_time(void);
+    void set_current_gps_time(lorawan_time_t gps_time);
+
+    /** Sets up UTC system time
+     *
+     * This API provides a convenience utility to setup UTC system time.
+     * Please note that device level synchronization does not require any conversion
+     * from GPS time. That's why any application level or stack level APIs involved
+     * in time synchronization should always use 'get_current_gps_time()' and
+     * 'set_current_gps_time(time)' APIs. 'set_system_time_utc(...)' API can be used
+     * for other application purposes where acquisition of UTC time is important.
+     * In addition to that it should be taken into account that the internal network
+     * assisted GPS time acquisition may not be 100% accurate. It involves local monotonic
+     * ticks (in ms) which is a direct function of CPU ticks and can be inaccurate. The
```

Comment:
Inaccuracy of the local monotonic ticks shouldn't really be an issue. (Certainly not relative to the fact that the RTC API is only 1 second resolution...)